src/stats.rs

```diff
@@ -102,6 +101,25 @@ impl ProjectStats {
 
         exceeds_budget
     }
+
+    /// Returns the spent budget, averaged *per-second*.
+    fn spent_budget(&self, now: Instant, truncated_now: Instant) -> f64 {
+        let earliest_time = truncated_now - self.config.budgeting_window;
+        let total_spent_budget: f64 = self
+            .budget_buckets
+            .iter()
+            .filter_map(|b| (b.0 >= earliest_time).then_some(b.1))
+            .sum();
+
+        // The configured budget is meant as a per-second budget.
+        // To calculate that, we want to divide by the real passed time,
+        // to avoid any artifacts resulting from the bucketing as much as possible.
+        let adjustment = now - truncated_now;
+        let adjusted_time_window =
+            self.config.budgeting_window - self.config.bucket_size + adjustment;
```

Comment:
This should make it work:
```suggestion
        let adjusted_time_window = if adjustment != Duration::ZERO {
            self.config.budgeting_window - self.config.bucket_size + adjustment
        } else {
            self.config.budgeting_window
        };
```